TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Alerts/AddAlert/Type.cshtml.cs

```diff
@@ -0,0 +1,84 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace TeachingRecordSystem.SupportUi.Pages.Alerts.AddAlert;
+
+[Journey(JourneyNames.AddAlert), RequireJourneyInstance]
+public class TypeModel(
+    TrsLinkGenerator linkGenerator,
+    ReferenceDataCache referenceDataCache) : PageModel
+{
+    public JourneyInstance<AddAlertState>? JourneyInstance { get; set; }
+
+    [FromQuery]
+    public Guid PersonId { get; set; }
+
+    [FromQuery]
+    public bool FromCheckAnswers { get; set; }
+
+    public string? PersonName { get; set; }
+
+    [BindProperty]
+    [Display(Name = "Select an alert type")]
+    [Required(ErrorMessage = "Select an alert type")]
+    public Guid? AlertTypeId { get; set; }
+
+    public AlertCategoryInfo[]? Categories { get; set; }
+
+    public AlertTypeInfo[]? AlertTypes { get; set; }
+
+    public async Task<IActionResult> OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        var selectedType = AlertTypes!.Single(t => t.AlertTypeId == AlertTypeId);
+
+        await JourneyInstance!.UpdateStateAsync(
+            state =>
+            {
+                state.AlertTypeId = AlertTypeId;
+                state.AlertTypeName = selectedType.Name;
+            });
+
+        return Redirect(FromCheckAnswers
+            ? linkGenerator.AlertAddCheckAnswers(PersonId, JourneyInstance.InstanceId)
+            : linkGenerator.AlertAddDetails(PersonId, JourneyInstance.InstanceId));
+    }
+
+    public async Task<IActionResult> OnPostCancel()
+    {
+        await JourneyInstance!.DeleteAsync();
+        return Redirect(linkGenerator.PersonAlerts(PersonId));
+    }
+
+    public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
+    {
+        var personInfo = context.HttpContext.GetCurrentPersonFeature();
+
+        PersonName = personInfo.Name;
+        AlertTypeId ??= JourneyInstance!.State.AlertTypeId;
+
+        var categories = await referenceDataCache.GetAlertCategories();
```

Comment:
To me it would be simpler to get the active alert types first, then get the categories for them (that way we avoid checking `IsActive` twice).